action.yaml

```diff
@@ -30,21 +36,85 @@ runs:
-        substitutions=""
-        for buildArg in ${buildArgs}; do
-          echo "--- $buildArg"
-          # the underscore and the comma are required to make it a valid Cloudbuild subsitution
-          substitutions="${substitutions}_${buildArg},"
-        done
-        echo $substitutions
+        set -x
+
+        context='.'
+        [[ "${{ inputs.context }}" != "" ]] && context=${{ inputs.context }}
+
+        if [[ "${{ inputs.dockerfile }}" != "" ]] || [[ "${{ inputs.context }}" != "" ]]; then
+          dockerfile=Dockerfile
+          [[ "${{ inputs.dockerfile }}" != "" ]] && dockerfile=${{ inputs.dockerfile }}
+
+          echo "Trigger Docker build for '${image}' using context '${context}' and Dockerfile '${dockerfile}': "
+          echo "(no support for build args)"
 
+          config=/tmp/cloubuild.yaml
+          substitutions="${substitutions}_DOCKERFILE=${dockerfile},"
+
+          registry_repository=$(cat ${context}/push_image)
+          version=$(cat ${context}/version)
+          image=${registry_repository}:${version}
+          cacheRepo=${registry_repository}/cache
+
+          cat <<EOF > /tmp/cloubuild.yaml
+        substitutions:
+          _IMAGE:
+          _CACHE_REPO:
+          _DOCKERFILE:
+        steps:
+          - name: 'gcr.io/kaniko-project/executor:v1.2.0'
+            args: # https://github.com/GoogleContainerTools/kaniko#additional-flags
+              - --destination=\$_IMAGE
+              - --cache-repo=\$_CACHE_REPO
+              - --dockerfile=\$_DOCKERFILE
+              - --cache=true
+              - --digest-file=/dev/termination-log
+        EOF
+        elif [[ "${{ inputs.config }}" != "" ]] && \
+             [[ "${{ inputs.image }}" != "" ]]; then
+          echo "Trigger Docker build for '${image}' using configuration '${{ inputs.config }}' and build args : "
+          buildArgs="${{ inputs.buildArgs }}"
+          substitutions=""
+          for buildArg in ${buildArgs}; do
+            echo "--- $buildArg"
+            # the underscore and the comma are required to make it a valid Cloudbuild subsitution
+            substitutions="${substitutions}_${buildArg},"
+          done
+          echo $substitutions
+
+          config="${{ inputs.config }}"
+          image="${{ inputs.image }}"
+          cacheRepo="${{ inputs.cacheRepo }}"
+
+        else
+          echo "config error"
+          exit 1
+        fi
+
+        echo "Using Gcloud config :"
+        cat ${config}
         set -x
-        gcloud --project zenhub-ops builds submit . --gcs-source-staging-dir=gs://ops-cloudbuild-bucket/data --gcs-log-dir=gs://ops-cloudbuild-bucket/logs \
-           --config ${{ inputs.config }} \
-           --substitutions _CACHE_REPO=${{ inputs.cacheRepo }},_IMAGE=${{ inputs.image }},${substitutions}
+        gcloud --project zenhub-ops builds submit ${context} --gcs-source-staging-dir=gs://ops-cloudbuild-bucket/data --gcs-log-dir=gs://ops-cloudbuild-bucket/logs \
```

Comment:
you could add `project` as an optional input and leave `zenhub-ops` ad default value